integration_tests/src/main/python/parquet_test.py

```diff
@@ -685,3 +686,339 @@ def test_parquet_reading_from_unaligned_pages_basic_filters_with_nulls(spark_tmp
+        explain = df._sc._jvm.PythonSQLUtils.explainString(df._jdf.queryExecution(), "simple")
+
+        assert "PushedAggregation: [MAX(id), MIN(id)]" in explain
+        return df
+
+    assert_gpu_and_cpu_are_equal_collect(do_explain, conf_for_parquet_aggregate_pushdown)
+
+
+@pytest.mark.skipif(is_before_spark_330(), reason='this is a new feature of Spark 330')
+@allow_non_gpu(any = True)
+def test_parquet_aggregate_push_down_different_data_types(spark_tmp_path):
+    data_path = spark_tmp_path + "/pushdown.parquet"
+    gen = StructGen([
+        ('string_c', string_gen),
+        ('bool_c', BooleanGen(False)), 
+        ('byte_c', ByteGen(False)),
+        ('short_c', ShortGen(False)),
+        ('int_c', int_gen), 
+        ('long_c', LongGen(False)),
+        ('float_c', FloatGen(False)), 
+        ('double_c', DoubleGen(False)),
+        ('decimal_c', DecimalGen(25, 5)),
+        ('date_c', DateGen(nullable= False)),
+        ('timestamp_c', timestamp_gen)], nullable= False
+    )
+
+    def timestamp_not_push_down(spark):
+
+        gen_df(spark, gen, 3).write.mode('overwrite').parquet(data_path)
+        
+        # INT96 sort order is not defined,  cannot be pushed down
+        # Parquet Binary min/max could be truncated, so we disable aggregate
+        # push down for Parquet Binary (could be Spark StringType, BinaryType or DecimalType).
+        df = spark.read.parquet(data_path).selectExpr(
+            "min(string_c)", "min(bool_c)", "min(byte_c)",
+              " min(short_c)", "min(int_c)", "min(long_c)", "min(float_c)",
+              "min(double_c)", "min(decimal_c)", "min(date_c)", "min(timestamp_c)")
+        explain = df._sc._jvm.PythonSQLUtils.explainString(df._jdf.queryExecution(), "simple")
+
+        assert  "PushedAggregation: []" in explain
+        return df
+
+    def without_timestamp_and_binary(spark):
+
+        gen_df(spark, gen, 3).write.mode('overwrite').parquet(data_path)
+        
+        df = spark.read.parquet(data_path).selectExpr(
+             "min(bool_c)", "min(byte_c)", " min(short_c)", "min(int_c)", "min(long_c)", "min(float_c)","min(double_c)", "min(date_c)")
+        explain = df._sc._jvm.PythonSQLUtils.explainString(df._jdf.queryExecution(), "simple")
+
+        assert  "PushedAggregation: [MIN(bool_c), MIN(byte_c), MIN(short_c), MIN(int_c), MIN(long_c), MIN(float_c), MIN(double_c), MIN(date_c)]" in explain
+        return df
+
+    conf = {"spark.sql.parquet.aggregatePushdown": "true", 
+            "spark.sql.sources.useV1SourceList": "",
+            "spark.sql.parquet.datetimeRebaseModeInWrite": "CORRECTED",
+            "spark.sql.parquet.int96RebaseModeInWrite": "CORRECTED",
+            "spark.rapids.sql.format.parquet.writer.int96.enabled": "false"}
+
+    
+    assert_gpu_and_cpu_are_equal_collect(timestamp_not_push_down, conf=conf)
+    assert_gpu_and_cpu_are_equal_collect(without_timestamp_and_binary, conf=conf)
```

Comment:
This looks like two unrelated tests crammed into one, as they have separate local functions being called.  Only significant thing being shared here is the data generator which can be factored out into a module variable and used in each separate test.